src/targets.rs

```diff
@@ -241,70 +396,132 @@ pub fn default_binary_format(triple: &Triple) -> BinaryFormat {
+            ArmArchitecture::Armv8_5a => "armv8.5a",
+            ArmArchitecture::Armv8mBase => "armv8m.base",
+            ArmArchitecture::Armv8mMain => "armv8m.main",
+            ArmArchitecture::Armv8r => "armv8r",
+            ArmArchitecture::Thumbv6m => "thumbv6m",
+            ArmArchitecture::Thumbv7a => "thumbv7a",
+            ArmArchitecture::Thumbv7em => "thumbv7em",
+            ArmArchitecture::Thumbv7m => "thumbv7m",
+            ArmArchitecture::Thumbv7neon => "thumbv7neon",
+            ArmArchitecture::Thumbv8mBase => "thumbv8m.base",
+            ArmArchitecture::Thumbv8mMain => "thumbv8m.main",
+            ArmArchitecture::Armebv7r => "armebv7r",
         };
         f.write_str(s)
     }
 }
 
+impl fmt::Display for Architecture {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if let Architecture::Arm(arm) = *self {
+            arm.fmt(f)
+        } else {
+            let s = match *self {
+                Architecture::Unknown => "unknown",
+                Architecture::Asmjs => "asmjs",
+                Architecture::I386 => "i386",
+                Architecture::I586 => "i586",
+                Architecture::I686 => "i686",
+                Architecture::Mips => "mips",
+                Architecture::Mips64 => "mips64",
+                Architecture::Mips64el => "mips64el",
+                Architecture::Mipsel => "mipsel",
+                Architecture::Msp430 => "msp430",
+                Architecture::Powerpc => "powerpc",
+                Architecture::Powerpc64 => "powerpc64",
+                Architecture::Powerpc64le => "powerpc64le",
+                Architecture::Riscv32 => "riscv32",
+                Architecture::Riscv32imac => "riscv32imac",
+                Architecture::Riscv32imc => "riscv32imc",
+                Architecture::Riscv64 => "riscv64",
+                Architecture::S390x => "s390x",
+                Architecture::Sparc => "sparc",
+                Architecture::Sparc64 => "sparc64",
+                Architecture::Sparcv9 => "sparcv9",
+                Architecture::Wasm32 => "wasm32",
+                Architecture::X86_64 => "x86_64",
+                Architecture::Arm(_) => unreachable!(),
+            };
+            f.write_str(s)
+        }
+    }
+}
+
 impl FromStr for Architecture {
     type Err = ();
 
     fn from_str(s: &str) -> Result<Self, ()> {
         Ok(match s {
             "unknown" => Architecture::Unknown,
-            "aarch64" => Architecture::Aarch64,
-            "arm" => Architecture::Arm,
-            "armebv7r" => Architecture::Armebv7r,
-            "armv4t" => Architecture::Armv4t,
-            "armv5te" => Architecture::Armv5te,
-            "armv6" => Architecture::Armv6,
-            "armv7" => Architecture::Armv7,
-            "armv7r" => Architecture::Armv7r,
-            "armv7s" => Architecture::Armv7s,
-            "armv7ve" => Architecture::Armv7ve,
+            "arm" => Architecture::Arm(ArmArchitecture::Arm),
```

Comment:
Would it work to implement `FromStr` for `ArmArchitecture`? We could do something like `_ if s.starts_with("arm") => ArmArchitecture::from_str(s)` and ditto for `"thumb"`.

Reply:
Implemented this a bit differently [here](https://github.com/CraneStation/target-lexicon/pull/13/commits/d34582bcf5941256d9fb896c45dc129064830a93#diff-ae890d647aba44a55c6cf4485817b171R620) - I'm not particularly thrilled about it - there's a possibility for multiple arches to try to handle the same string - but this allows `Aarch64` to handle something like `"arm64"`.

Reply:
Ok. I think the approach you took here is fine for now. We're pretty conservative about adding new strings, so we're unlikely to have multiple arches trying to grab the same string. And if we add a lot more architecture families, we can figure out a different approach at that time.